app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -2422,6 +2426,19 @@ class BrowserTabViewModel @Inject constructor(
         command.value = HandleNonHttpAppLink(appLink, getUrlHeaders(appLink.fallbackUrl))
     }
 
+    fun onPrintSelected() {
+        url?.let {
+            pixel.fire(AppPixelName.MENU_ACTION_PRINT_PRESSED)
+            command.value = PrintLink(removeAtbAndSourceParamsFromSearch(it))
+        }
+    }
+
+    fun printFromWebView() {
+        viewModelScope.launch(dispatchers.main()) {
```

Comment:
The dispatcher can be omitted here as main is the default in viewModelScope. Also. you can call `onPrintSelected()` directly, without launching a coroutine.

Reply:
Yeah, it doesn't work unless you change to the main thread. Thanks a lot for the comment, I'm not very experienced with coroutines so I didn't know main was the default one and can be omitted 🙌